includes/promoted-jobs/class-wp-job-manager-promoted-jobs-status-handler.php

```diff
@@ -102,4 +126,47 @@ private function get_site_feed_url() {
 		return add_query_arg( 'site', home_url( '', 'https' ), WP_Job_Manager_Helper_API::get_wpjmcom_url() . '/wp-json/promoted-jobs/v1/site/jobs' );
 	}
 
+	/**
+	 * Updates the interval option with the value from the response header from the feed.
+	 *
+	 * @param array|\WP_Error $response The HTTP response from the feed.
+	 * @param string          $header_name The name of the header to check for the interval value.
+	 * @param string          $option_name The name of the option to update with the interval value.
+	 * @return void
+	 */
+	private function update_interval( $response, $header_name, $option_name ) {
+		$headers = wp_remote_retrieve_headers( $response );
+		if ( isset( $headers[ $header_name ] ) ) {
+			$header = $headers [ $header_name ];
+			if ( 'false' === $header ) {
+				$header = '0';
+			} elseif ( ! rest_is_integer( $header ) ) {
+				return;
+			}
+			$value = (int) $header;
+			if ( self::WEBHOOK_INTERVAL_OPTION_KEY === $option_name && 0 === $value ) {
+				// If the webhook interval is zero, we don't remove the option, to not block webhook updates.
+				return;
+			}
+			if ( 0 === $value ) {
+				// If the interval is zero, we remove the option, because it is the default anyway.
+				delete_option( $option_name );
+			} else {
+				update_option( $option_name, $value, false );
+			}
+		}
+	}
+
+	/**
+	 * Initialize the default options for the Promoted Jobs Status Handler.
+	 *
+	 * @return void
+	 */
+	public static function initialize_defaults() {
+		$webhook_interval = 5 * MINUTE_IN_SECONDS;
+		update_option( self::LAST_CHECK_OPTION_KEY, time() - $webhook_interval, '', false );
```

Comment:
Just one final thing I noticed:
Since we're handling the webhook interval dynamically, I think we should get this interval from the option and use this only if the option doesn't exist, right?

Reply:
I implemented your suggestion in 4e0b5a1. Thanks!